pkg/handlers/generic/lifecycle/csi/aws-ebs/handler.go

```diff
@@ -0,0 +1,132 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/k8s/client"
+	"github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	crsv1 "sigs.k8s.io/cluster-api/exp/addons/api/v1beta1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type AWSEBSConfig struct {
+	defaultsNamespace          string
+	defaultAWSEBSConfigMapName string
+}
+
+func (a *AWSEBSConfig) AddFlags(prefix string, flags *pflag.FlagSet) {
+	flags.StringVar(
+		&a.defaultsNamespace,
+		prefix+".defaultsNamespace",
+		corev1.NamespaceDefault,
+		"namespace of the ConfigMap used to deploy AWS EBS CSI driver",
+	)
+	flags.StringVar(
+		&a.defaultAWSEBSConfigMapName,
+		prefix+".aws-ebs-provider-configmap-name",
+		"aws-ebs-crs-cm",
+		"name of the ConfigMap used to deploy AWS EBS CSI driver",
+	)
+}
+
+type AWSEBS struct {
+	client ctrlclient.Client
+	config *AWSEBSConfig
+}
+
+func New(
+	c ctrlclient.Client,
+	cfg *AWSEBSConfig,
+) *AWSEBS {
+	return &AWSEBS{
+		client: c,
+		config: cfg,
+	}
+}
+
+func (a *AWSEBS) EnsureCSIConfigMapForCluster(ctx context.Context, cluster *clusterv1.Cluster) (*corev1.ConfigMap, error) {
```

Comment:
How about we move this into the main CSI handler or some other shared package? And here we can have functions that return driver details.

Things like:
```
func GetName()
func GetConfigMapName()
```

Reply:
Maybe `GetName` and `EnsureConfigMap` would be good. I suspect that the code for creating the CRS would be very similar for all of them, so that code can live on the handler level

Reply:
I don't even think we need `GetName` now that I think about it more. The user will provide a map to the CSIHandler which maps names to CSI providers